services/profile/profileServices.ts

```diff
@@ -0,0 +1,27 @@
+import { AxiosResponse } from "axios";
+import api from "../../shared/fetcher";
+
+export interface ProfileResponse {
+  user_id: string;
+  country_code: string;
+  username: string;
+  profile_pic_path: string;
+  profile_pic_asset_id: null;
+  profile_pic_public_id: null;
+  bio: null;
+  is_anonymous: boolean;
+  allow_anon_dm: boolean;
+  only_received_dm_from_user_following: boolean;
+  verified_status: string;
+  following_symbol: string;
+  follower_symbol: string;
+}
+
+const getPublicProfile = async (username: string) => {
+  const { data } = await api.get<AxiosResponse<ProfileResponse>>(
+    `/api/v1/profiles/get-profile-public/${username}`
+  );
+  return data.data;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getPublicProfile` does not have any error handling. If the API request fails for any reason, it will throw an unhandled promise rejection. It would be better to wrap the API call in a try-catch block and handle any potential errors.
```diff
const getPublicProfile = async (username: string) => {
  try {
    const { data } = await api.get<AxiosResponse<ProfileResponse>>(
      `/api/v1/profiles/get-profile-public/${username}`
    );
    return data.data;
+    } catch (error) {
+      console.error(`Error fetching public profile for ${username}:`, error);
+      throw error;
+    }
};
```
This way, if there's an issue with the API call, it will be logged and re-thrown for further handling by the caller.


<!-- This is an auto-generated comment by OSS CodeRabbit -->